cli/src/revset_util.rs

```diff
@@ -213,6 +214,37 @@ pub fn parse_immutable_heads_expression(
     Ok(heads.union(&RevsetExpression::root()))
 }
 
+/// Prints warning if `trunk()` alias cannot be resolved. This alias could be
+/// generated by `jj git init`/`clone`.
+pub(super) fn warn_unresolvable_trunk(
+    ui: &Ui,
+    repo: &dyn Repo,
+    context: &RevsetParseContext,
+) -> io::Result<()> {
+    let (_, _, revset_str) = context
+        .aliases_map()
+        .get_function("trunk", 0)
+        .expect("trunk() should be defined by default");
+    let Ok(expression) = revset::parse(&mut RevsetDiagnostics::new(), revset_str, context) else {
+        // Parse error would have been reported.
+        return Ok(());
+    };
+    // Not using IdPrefixContext since trunk() revset shouldn't contain short
+    // prefixes.
+    let symbol_resolver = DefaultSymbolResolver::new(repo, context.symbol_resolvers());
+    if let Err(err) = expression.resolve_user_expression(repo, &symbol_resolver) {
+        writeln!(
+            ui.warning_default(),
+            "Failed to resolve `revset-aliases.trunk()`: {err}"
+        )?;
+        writeln!(
+            ui.hint_default(),
+            "Use `jj config edit --repo` to update or delete `trunk()` alias."
```

Comment:
Nit: s/update or delete/adjust the/ since that implies an action of either updating or deleting. But that is mostly preference.